### DISCUSSION TITLE
Downgrading versions of terraform

### DISCUSSION THREAD

**Discussion by pete0emerson** [gruntwork-io/knowledge-base#111] [Help] [answered]
We upgraded our openvpn infra to terraform 1.0.X, but we have an RDS module that is at 0.12.X and it uses state from the openvpn infra.  what is the preferred method to downgrade the terraform version of an existing piece of infra?  We can destroy it, remove the state file, and the DynamoDB entry, or revert to an previous version of the state file, and adjust the dynamo DB s3 object version, right?  is there a better way?  is one of these methods preferred?

**Comment by gruntwork-discussions** [comment]
You must label this issue with at least one of the following labels before it can be answered:

* [r:terraform-aws-service-catalog](https://github.com/gruntwork-io/terraform-aws-service-catalog)
* [r:terraform-aws-vpc](https://github.com/gruntwork-io/terraform-aws-vpc)
* [r:terraform-aws-architecture-catalog](https://github.com/gruntwork-io/terraform-aws-architecture-catalog)
* [r:terraform-aws-asg](https://github.com/gruntwork-io/terraform-aws-asg)
* [r:terraform-aws-beanstalk](https://github.com/gruntwork-io/terraform-aws-beanstalk)
* [r:terraform-aws-cache](https://github.com/gruntwork-io/terraform-aws-cache)
* [r:terraform-aws-ci](https://github.com/gruntwork-io/terraform-aws-ci)
* [r:terraform-aws-ci-pipeline-example](https://github.com/gruntwork-io/terraform-aws-ci-pipeline-example)
* [r:terraform-aws-cis-service-catalog](https://github.com/gruntwork-io/terraform-aws-cis-service-catalog)
* [r:terraform-aws-data-storage](https://github.com/gruntwork-io/terraform-aws-data-storage)
* [r:terraform-aws-ecs](https://github.com/gruntwork-io/terraform-aws-ecs)
* [r:terraform-aws-eks](https://github.com/gruntwork-io/terraform-aws-eks)
* [r:terraform-aws-elk](https://github.com/gruntwork-io/terraform-aws-elk)
* [r:terraform-aws-kafka](https://github.com/gruntwork-io/terraform-aws-kafka)
* [r:terraform-aws-lambda](https://github.com/gruntwork-io/terraform-aws-lambda)
* [r:terraform-aws-load-balancer](https://github.com/gruntwork-io/terraform-aws-load-balancer)
* [r:terraform-aws-messaging](https://github.com/gruntwork-io/terraform-aws-messaging)
* [r:terraform-aws-mongodb](https://github.com/gruntwork-io/terraform-aws-mongodb)
* [r:terraform-aws-monitoring](https://github.com/gruntwork-io/terraform-aws-monitoring)
* [r:terraform-aws-openvpn](https://github.com/gruntwork-io/terraform-aws-openvpn)
* [r:terraform-aws-sam](https://github.com/gruntwork-io/terraform-aws-sam)
* [r:terraform-aws-security](https://github.com/gruntwork-io/terraform-aws-security)
* [r:terraform-aws-server](https://github.com/gruntwork-io/terraform-aws-server)
* [r:terraform-aws-static-assets](https://github.com/gruntwork-io/terraform-aws-static-assets)
* [r:terraform-aws-zookeeper](https://github.com/gruntwork-io/terraform-aws-zookeeper)
* [r:terragrunt](https://github.com/gruntwork-io/terragrunt)
* [r:terratest](https://github.com/gruntwork-io/terratest)
* [r:repo-copier](https://github.com/gruntwork-io/repo-copier)


[Community Guidelines](https://github.com/gruntwork-io/knowledge-base/discussions/1)
[Getting help](https://gruntwork.io/support/)
[Gruntwork documentation](https://www.gruntwork.io)

**Comment by pete0emerson** [comment, marked as answer]
Your suggestions are perfectly reasonable. If the configuration of your OpenVPN infrastructure hasn't changed between the old 0.12.x and the new 1.0.X, I think it's simplest to revert to the previous version of the state file and adjust the dynamoDB S3 object accordingly:

1) Revert state file
2) Adjust dynamoDB S3 object
3) Run a terraform-0.12.x plan on the OpenVPN infrastructure and make sure it runs clean

If you have made modifications or the tf plan (with version 0.12.x) doesn't run cleanly, then you can take the next more drastic step of rebuilding the OpenVPN infra. Less desirable, obviously, due to downtime.
